### PR TITLE
Feat/test loger mixin

### PR DESCRIPTION
**Pull request by MementoRC**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
FIX for: https://github.com/hummingbot/hummingbot/pull/6383

Looks like I did not know how to work this update in. There is a code fix and a feature:
   - FIX: The `log_level` was not correctly set to `1` when the loggers are added
       * It works for the tests, but not in a case seen in MQTT test suite
   - FIX: Some of the tests were testing the wrong thing (this is why the flaw above passed the test)
   - FEAT: Added klpanagi `wait_for_logged()` method when dealing with async logging of server (for example)

**Tests performed by the developer**:
Passed the tests. Passed the failing tests when used with MQTT test suite

**Tips for QA testing**:

### CODE REVIEW

**Comment by MementoRC** [comment]
@cardosofede Please review, the `TestLoggerMixin` has a flaw

File: hummingbot/VERSION [hunk 1]
```diff
@@ -1 +1,2 @@
-dev-1.18.0
+
+1.17.0
```
Comment (cardosofede) [inline hummingbot/VERSION:2]: please fix the version

**Review by cardosofede — CHANGES_REQUESTED** [review]

File: test/test_logger_mixin.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
+import asyncio
 import unittest
 from logging import Logger, LogRecord
 from test.logger_mxin import LogLevel, TestLoggerMixin
 
-from hummingbot.logger import HummingbotLogger
+from hummingbot.logger.logger import HummingbotLogger
```
Comment (cardosofede) [inline test/test_logger_mixin.py:6]: why you are adding an extra .logger

**Review by cardosofede — CHANGES_REQUESTED** [review]

File: test/test_logger_mixin.py (hunk 2 above)
Comment (MementoRC) [reply test/test_logger_mixin.py]: HummingbotLogger is defined in `hummingbot/logger/logger.py`. Otherwise it is underlined red in Pycharm

**Comment by MementoRC** [comment]
@cardosofede Can you please review, I have made some changes:
   - The name seems more appropriate now: LoggerMixinForTest (open to suggestions)
   - More pythonic coding

**Review by cardosofede — APPROVED** [review]
LGTM!

**Pull request merged**